std/node/string_decoder.js

```diff
@@ -0,0 +1,257 @@
+//TODO@Soremwar
+//Definetely stilize this
+//Add tests
+//Add types
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+import { Buffer } from "./buffer.ts";
+
+const isEncoding =
+  Buffer.isEncoding ||
+  function (encoding) {
+    encoding = "" + encoding;
+    switch (encoding && encoding.toLowerCase()) {
+      case "hex":
+      case "utf8":
+      case "utf-8":
+      case "base64":
+      case "raw":
+        return true;
+      default:
+        return false;
+    }
+  };
+
+function _normalizeEncoding(enc) {
```

Comment:
We should use https://github.com/denoland/deno/blob/2fe315bfb1f8fc80e7e4e353414eb9bfa58c059f/std/node/_utils.ts#L52

And throw `notImplemented` not implemented encodings, to prevent having `normalizeEncoding` in every module

Reply:
Just added